plugins/regex_revalidate/regex_revalidate.c

```diff
@@ -414,19 +496,20 @@ main_handler(TSCont cont, TSEvent event, void *edata)
       if (status == TS_CACHE_LOOKUP_HIT_FRESH) {
         pstate = (plugin_state_t *)TSContDataGet(cont);
         iptr   = pstate->invalidate_list;
-        while (iptr) {
+        while (iptr && iptr->expiry >= now) {
+          // only unexpired lines (from front)
           if (!date) {
             date = get_date_from_cached_hdr(txn);
             now  = time(NULL);
           }
-          if ((difftime(iptr->epoch, date) >= 0) && (difftime(iptr->expiry, now) >= 0)) {
+          if (iptr->refresh >= date) {
```

Comment:
date is time_t so it must match a uint64_t if the system supports it. a double-conversion isn't needed.